conan/external_tools/vswhere_test.py

```diff
@@ -0,0 +1,169 @@
+
+        # Detect only one product (VS Community 15) as vswhere default detection
+        nproducts = len(tools.vswhere())
+        self.assertEqual(nproducts, self.modern_products)
+
+        # Detect only modern products (VS Community 15 & BuildTools 15)
+        products = tools.vswhere(products=["*"])
+        nproducts = len(products)
+        
+        self.assertEqual(nproducts, self.all_modern_products)
+        for product in products:
+            install_path = product["installationPath"]
+            self.assertTrue("Community" in install_path or
+                            "BuildTools" in install_path)
+
+        # Detect also legacy products but no modern BuildTools
+        products = tools.vswhere(legacy=True)
+        nproducts = len(products)
+
+        self.assertEqual(nproducts, self.modern_and_legacy_products)
+        for product in products:
+            install_path = product["installationPath"]
+            self.assertTrue("Community" in install_path or
+                            "Microsoft Visual Studio 14.0" in install_path)
+
+        # Detect all installed products
+        products = tools.vswhere(products=["*"])
+        products += tools.vswhere(legacy=["*"])
+        seen_products = []
+        for product in products:
+            if product not in seen_products:
+                seen_products.append(product)
+        products = seen_products
+        nproducts = len(products)
+
+        self.assertEqual(nproducts, self.all_products)
+        for product in products:
+            install_path = product["installationPath"]
+            self.assertTrue("Community" in install_path or
+                            "BuildTools" in install_path or
+                            "Microsoft Visual Studio 14.0" in install_path)
+
+    def vs_installation_path_test(self):
+        # Default behaviour
+        install_path = tools.vs_installation_path("15")
+        self.assertIn("Community", install_path)
+        install_path = tools.vs_installation_path("14")
+        self.assertIn("Microsoft Visual Studio 14.0", install_path)
+
+        # only BuildTools detection
+        install_path = tools.vs_installation_path("15", preference=["BuildTools"])
+        self.assertIn("BuildTools", install_path)
+        install_path = tools.vs_installation_path("14", preference=["BuildTools"])
+        self.assertIn("Microsoft Visual Studio 14.0", install_path)
+
+        # Ask for not installed versions
+        install_path = tools.vs_installation_path("15", preference=["Enterprise"])
+        self.assertIsNone(install_path)
+        install_path = tools.vs_installation_path("15", preference=["Professional"])
+        self.assertIsNone(install_path)
+
+        # Change preference order
+        install_path = tools.vs_installation_path("15", preference=["BuildTools", "Community", "Professional", "Enterprise"])
+        self.assertNotIn("Community", install_path)
```

Comment:
It is really needed all this assertNotIn? if there is only 1 path, and you know which one should be... is it correct to just check the assertIn?

Reply:
OK, I will remove `assertNotIn()`